addons/mail/models/mail_thread.py

```diff
@@ -1880,6 +1880,24 @@ def _message_post_after_hook(self, message, msg_vals):
         using already-computed values instead of having to rebrowse things. """
         pass
 
+    @api.model
+    def _message_edit_content(self, message, body):
+        """
+        Edits the content of a message.
+        """
+        user = self.env.user
+        can_edit = user.partner_id == message.author_id or user.has_group('base.group_system')
+        if message.model == 'mail.channel':
```

Comment:
Still not modular ^^ . Probably the check access method will also have to be split (_message_edit_content_check that is overridden). A bit verbose but allow to have a modular implementation.